src/intervals/arithmetic/basic.jl

```diff
@@ -0,0 +1,302 @@
+
+    if zero(T) ∈ a
+        a.lo < zero(T) == a.hi && return @round(F, T(-Inf), inv(a.lo))
+        a.lo == zero(T) < a.hi && return @round(F, inv(a.hi), T(Inf))
+        a.lo < zero(T) < a.hi && return entireinterval(F)
+        isthinzero(a) && return div_by_thin_zero(one(F))
+    end
+
+    return @round(F, inv(a.hi), inv(a.lo))
+end
+
+# Rational division
+//(a::Interval, b::Interval) = a / b
+
+function min_ignore_nans(args...)
+    min(Iterators.filter(x->!isnan(x), args)...)
+end
+
+function max_ignore_nans(args...)
+    max(Iterators.filter(x->!isnan(x), args)...)
+end
+
+"""
+    fma(a::Interval, b::Interval, c::Interval)
+
+Fused multiply-add.
+
+Implement the `fma` function of the IEEE Std 1788-2015 (Table 9.1).
+"""
+function fma(a::F, b::F, c::F) where {T, F<:Interval{T}}
+    (isempty(a) || isempty(b) || isempty(c)) && return emptyinterval(F)
+
+    if isentire(a)
+        isthinzero(b) && return c
+        return entireinterval(F)
+    elseif isentire(b)
+        isthinzero(a) && return c
+        return entireinterval(F)
+    end
+
+    lo = setrounding(T, RoundDown) do
+        lo1 = fma(a.lo, b.lo, c.lo)
+        lo2 = fma(a.lo, b.hi, c.lo)
+        lo3 = fma(a.hi, b.lo, c.lo)
+        lo4 = fma(a.hi, b.hi, c.lo)
+        min_ignore_nans(lo1, lo2, lo3, lo4)
+    end
+
+    hi = setrounding(T, RoundUp) do
+        hi1 = fma(a.lo, b.lo, c.hi)
+        hi2 = fma(a.lo, b.hi, c.hi)
+        hi3 = fma(a.hi, b.lo, c.hi)
+        hi4 = fma(a.hi, b.hi, c.hi)
+        max_ignore_nans(hi1, hi2, hi3, hi4)
+    end
+
+    return F(lo, hi)
+end
+
+"""
+    sqrt(a::Interval)
+
+Square root of an interval.
+
+Implement the `sqrt` function of the IEEE Std 1788-2015 (Table 9.1).
+"""
+function sqrt(a::F) where {F<:Interval}
+    domain = F(0, Inf)
+    a = a ∩ domain
+
+    isempty(a) && return a
+
+    return @round(F, sqrt(a.lo), sqrt(a.hi))  # `sqrt` is correctly-rounded
+end
+
+
+"""
+    nthroot(a::Interval{BigFloat}, n::Integer)
+
+Compute the real n-th root of Interval.
+"""
+function nthroot(a::Interval{BigFloat}, n::Integer)
+    n == 1 && return a
+    n < 0 && isthinzero(a) && return emptyinterval(a)
+    isempty(a) && return a
+    if n > 0
+        a.hi < 0 && iseven(n) && return emptyinterval(BigFloat)
+        if a.lo < 0 && a.hi >= 0 && iseven(n)
+            a = a ∩ Interval{BigFloat}(0, Inf)
+        end
+        ui = convert(Culong, n)
+        low = BigFloat()
+        high = BigFloat()
+        ccall((:mpfr_rootn_ui, :libmpfr), Int32 , (Ref{BigFloat}, Ref{BigFloat}, Culong, MPFRRoundingMode) , low , a.lo , ui, MPFRRoundDown)
+        ccall((:mpfr_rootn_ui, :libmpfr), Int32 , (Ref{BigFloat}, Ref{BigFloat}, Culong, MPFRRoundingMode) , high , a.hi , ui, MPFRRoundUp)
+        b = interval(low , high)
+        return b
+    elseif n < 0
+        return inv(nthroot(a, -n))
+    elseif n == 0
+        return emptyinterval(a)
+    end
+end
+
+function nthroot(a::Interval{T}, n::Integer) where T
+    b = nthroot(big(a), n)
```

Comment:
We should add here the special cases `n==1` and `n==2` as well; my guess is that it will improve performance.